meta-lmp-base/recipes-security/softhsm/softhsm_2.%.bbappend

```diff
@@ -7,6 +7,7 @@ SRC_URI:append = " \
 do_install:append() {
 	if ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'true', 'false', d)}; then
 		install -D -m 0644 ${WORKDIR}/tmpfiles.conf ${D}${nonarch_libdir}/tmpfiles.d/softhsm.conf
+		rm -r ${D}${localstatedir}
```

Comment:
imo `rmdir` is more safe because with that we control what is removed and when something new appears we will be notified with a new error.

```
(cd ${D}${localstatedir}; rmdir -v --parents ./lib/softhsm/tokens)
```
I have tried that  and it removes the full tree.

```
+ cd /build/tmp-lmp/work/corei7-64-lmp-linux/softhsm/2.6.1-r0/image/var
+ rmdir -vp ./lib/softhsm/tokens

rmdir: removing directory, './lib/softhsm/tokens'
rmdir: removing directory, './lib/softhsm'
rmdir: removing directory, './lib'
rmdir: removing directory, '.'
```

Reply:
Sure, tried to use rmdir first but you have to get into the main directory first, which annoyed me a bit :-)

But I agree that it is better on the long run, will update.

Reply:
Done